### PR TITLE
Show active bookings & sort by date

### PR DESCRIPTION
**Pull request by Sajaas**
## PT: https://www.pivotaltracker.com/story/show/151582988

* Only showing active bookings
* Listing the bookings by attribute :time

### CODE REVIEW

**Review by tochman — CHANGES_REQUESTED** [review]
I see no acceptance tests. ???

**Review by tochman — COMMENTED** [review]
In what was is the user restricted to only be able to book 2 slots? By being told so?

**Comment by tochman** [comment]
Ah, wrong PR. I understand. But you still need a test for the changed behavior introduced by this pr

File: features/show_bookings.feature [hunk 1]
```diff
@@ -16,4 +16,10 @@ Feature: User can see a list of their laundry bookings
...
   Scenario: User don't have any bookings
     Given user visits the booking page
-    Then I should see the text "You don't have any bookings yet."
\ No newline at end of file
+    Then I should see the text "You don't have any bookings yet."
+
+  Scenario: User should not see passed bookings
+    Given user visits the booking page
+    And I click on calendar link "01 Sep 08:00"
+    Then time is frozen at 2017-09-02
```
Comment (tochman) [inline features/show_bookings.feature]: You need to reload the page by visiting it again.

Comment (tochman) [reply features/show_bookings.feature]: Also, change `Then` -> `And`

Comment (Sajaas) [reply features/show_bookings.feature]: THANK YOU! 😭

File: features/step_definitions/laundry_room_steps.rb [hunk 2]
```diff
@@ -35,4 +35,16 @@
...
+Given("there is a booking from for {string}") do |date|
+  slot = Time.zone.parse(date)
+  slot_to_book = @laundry.schedule
+                     .occurrences(1.week.from_now)
+                     .detect { |occ| occ.beginning_of_hour.localtime == slot.localtime }
+  @user.book! @laundry, time: slot_to_book, amount: 1
+end
+
+Then(/^I should not see the text "([^"]*)"$/) do |text|
+  page.should have_no_content(text)
```
Comment (tochman) [inline features/step_definitions/laundry_room_steps.rb]: Hmm.. Please restrain from too much cut and paste. This assertion is using a syntax that we have not used in the camp and that is a few years old. We use `expect` in our assertions.

File: features/step_definitions/laundry_room_steps.rb [hunk 3]
```diff
@@ -35,4 +35,16 @@
...
   visit laundry_room_create_booking_path(@laundry, time: date)
+end
+
+Given("there is a booking from for {string}") do |date|
+  slot = Time.zone.parse(date)
+  slot_to_book = @laundry.schedule
+                     .occurrences(1.week.from_now)
+                     .detect { |occ| occ.beginning_of_hour.localtime == slot.localtime }
+  @user.book! @laundry, time: slot_to_book, amount: 1
+end
```
Comment (tochman) [inline features/step_definitions/laundry_room_steps.rb]: I don't understand. There should be a step like this in our step definitions. I can't see any reason to create a new one.

Comment (Sajaas) [reply features/step_definitions/laundry_room_steps.rb]: Yeah, that's accidental leftovers from when I tried to book a laundry room before 1/9-17. Deleting it.

**Review by tochman — APPROVED** [review]

**Pull request merged**